File: frontend/src/objects/merchant/MerchantBusinessHours.ts
export type MerchantBusinessStatus = 'open' | 'resting' | 'closedToday' | 'paused'

export interface MerchantWeeklyBusinessHour {
  dayOfWeek: number
  startTime: string
  endTime: string
  enabled: boolean
}

export interface MerchantHolidayBusinessHour {
  date: string
  businessStatus: MerchantBusinessStatus
  startTime?: string | null
  endTime?: string | null
}

File: backend/src/merchant/objects/MerchantBusinessHours.scala
package delivery.merchant.objects

final case class MerchantWeeklyBusinessHour(
    dayOfWeek: Int,
    startTime: String,
    endTime: String,
    enabled: Boolean = true
)

final case class MerchantHolidayBusinessHour(
    date: String,
    businessStatus: String,
    startTime: Option[String] = None,
    endTime: Option[String] = None
)
